### PR TITLE
Add Marian batch I/O contract tests

### PR DESCRIPTION
**Pull request by mustjab**
Adds two C API tests backed by dummy Marian models covering the batch and beam tensor shapes fixed in #2337.

- `MarianBatchIOContract`: batch 2, single beam
- `MarianBatchWithBeamsIOContract`: batch 2, two beams

The models declare fixed input dimensions, so a wrongly sized batch or beam tensor fails to bind. Prompt length is chosen so the beam case is discriminating: with 2-token prompts the correct encoder width is 3 and the buggy one is 4, whereas 1-token prompts give 2 either way.

Fixtures are generated by `test/python/create/create_marian_batch_models.py`. Outputs use `ConstantOfShape` so the committed models stay under 1 KB each.

**Coverage:** each fix from #2337 was reverted in turn to confirm the tests catch it. They detect tensor sizing regressions (decoder primed with one bos instead of one per batch*beam; eos appended once at the end of the flattened buffer instead of per row). They do not detect value-only regressions such as attention mask contents, since the graphs compute nothing.

### CODE REVIEW

File: test/python/create/create_marian_batch_models.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                TensorProto.FLOAT,
+                [batch_beam, seq, HIDDEN_SIZE],
+            ),
+            ("encoder_attention_mask", TensorProto.INT32, [batch_beam, seq]),
+            (
+                "rnn_states_prev",
+                TensorProto.FLOAT,
+                [RNN_STATE_SLOTS, batch_beam, HIDDEN_SIZE],
+            ),
+            ("past_key_values_length", TensorProto.INT32, [1]),
```
[the remark this answers is not in the document]
Comment (mustjab) [reply test/python/create/create_marian_batch_models.py]: Good catch, fixed in 3d4c43a3. Confirmed the decoder binds `past_key_values_length` as a rank-0 int64, so the fixture declared both the wrong type and the wrong rank; the int32 tensor was being written through `GetTensorMutableData<int64_t>()`, overrunning it by 4 bytes. Also corrected `input_ids` to int64 on both graphs, which additionally exercises the cast path in `MarianInputIDs::Update` that the int32 fixture was skipping.

File: test/c_api_tests.cpp [hunk 2]
```diff
@@ -416,6 +416,50 @@ TEST(CAPITests, EndToEndPhiBatch) {
 #endif
 }
 
+// Marian sizes encoder buffers by batch_size and decoder buffers by
+// batch_size * num_beams. These fixtures are shapes-only graphs, so they pin
+// that arithmetic and nothing about generated text. Each configuration needs
+// its own fixture because dummy graphs cannot be shape-polymorphic.
```
[the remark this answers is not in the document]
Comment (mustjab) [reply test/c_api_tests.cpp]: Agreed, the comment was wrong. Reworded in 3d4c43a3 to state that every ORT tensor is batch*beam wide: `MarianState::Run` receives a `[batch_size, sequence]` prompt buffer and `DefaultInputIDs` expands it over beams, so the batch_size/batch*beam split applies to the prompt buffer, not to encoder vs decoder.

**Comment by mustjab** [comment]
Addressed the suppressed review note about `past_key_values_length` rank in 7b9e31f8.

The stated failure mode does not occur: rank-0 loads and runs fine, because a rank-0 declaration carries no dimension constraints for ORT to check. Verified by declaring it `[5]`, which does fail with `Got: 1 Expected: 5`, so the input is validated when constraints exist.

Changed it to `[1]` anyway, for a different reason: rank-0 asserted nothing about this input, whereas `[1]` pins the single element `MarianState` binds and would catch a future change that sized it per batch entry. Re-confirmed the fixtures still fail when the beam fix is reverted.

**Review by kunal-vaishnavi — APPROVED** [review]
